library/modules/Gui.cpp

```diff
@@ -1360,6 +1368,100 @@ DFHACK_EXPORT void Gui::writeToGamelog(std::string message)
     fseed.close();
 }
 
+namespace
+{   // Utility functions for reports
+    bool parseReportString(std::vector<std::string> *out, const std::string &str, size_t line_length = 73)
+    {   // parse a string into output strings like DF does for reports
+        if (str.empty() || line_length == 0)
+            return false;
+
+        string parsed;
+        size_t i = 0;
+
+        do
+        {
+            if (str[i] == '&') // escape character
+            {
+                i++; // ignore the '&' itself
+                if (i >= str.length())
+                    break;
+
+                if (str[i] == 'r') // "&r" adds a blank line
+                {
+                    word_wrap(out, parsed, line_length, false/*, true*/);
+                    out->push_back(" "); // DF adds a line with a space for some reason
+                    parsed.clear();
+                }
+                else if (str[i] == '&') // "&&" is '&'
+                    parsed.push_back('&');
+                // else next char is ignored
+            }
+            else
+                parsed.push_back(str[i]);
+        }
+        while (++i < str.length());
+
+        if (parsed.length())
+            word_wrap(out, parsed, line_length, false/*, true*/);
+
+        return true;
+    }
+
+    bool recent_report(df::unit *unit, df::unit_report_type slot)
+    {
+        if (unit && !unit->reports.log[slot].empty() &&
+            *df::global::cur_year == unit->reports.last_year[slot] &&
+            (*df::global::cur_year_tick - unit->reports.last_year_tick[slot]) <= 500)
```

Comment:
what is the significance of `500`? Can we make it a named constant?

Reply:
It's the number of ticks DF considers a report to be "recent".

Reply:
IMO wouldn't hurt to have a constant for that. If dealing with a `coord` or `coord2d`, we could use its `isValid()` method instead of a raw check.

Reply:
@lethosor
The ``isValid()`` check itself uses the magic number. It's all over plugins too. I'm not sure of the best place to define it, but it's probably out of the scope of this PR.